### ISSUE TITLE
Google Music support

### ISSUE THREAD

**Issue by Clementine-Issue-Importer** [clementine-player/Clementine#2204]
_From [FabioAr...@gmail.com](https://code.google.com/u/103706141194269180730/) on September 01, 2011 18:11:56_

Can you make possible listen music saved in google music account in clementine?

_Original issue: http://code.google.com/p/clementine-player/issues/detail?id=2204_

**Comment by malexandre** [comment]
+1. Any news for this? If there is no plan on doing this, I might find the time to try it, even though C++ is not my primary language anymore. But I don't want to duplicate the work :D.

**Comment by robin92** [comment]
@malexandre 
If you know any library that could connect to Google Music, I would help you in implementing this.

**Comment by liftarn** [comment]
@robin92 There is an unofficial api https://github.com/simon-weber/Unofficial-Google-Music-API

**Comment by ajohnston9** [comment]
@robin92 @malexandre I'd help also; we can get a fork going.

**Comment by lesthack** [comment]
+1.  News ?

**Comment by JonasCz** [comment]
... +1

**Comment by Ziggoto** [comment]
+ 1

**Comment by eligrey** [comment]
+1

**Comment by ErLitri** [comment]
+1.

**Comment by morsik** [comment]
+1
Just tried Tomahawk… didn't like it :<

**Comment by rickinbsb** [comment]
+1

**Comment by riyadparvez** [comment]
+1

**Comment by isality** [comment]
+1

**Comment by zeachco** [comment]
+1

**Comment by marek287** [comment]
+1

**Comment by rmfought** [comment]
+1

**Comment by pschachte** [comment]
+1

**Comment by eligrey** [comment]
This is only tangentially related, but if any of you want a quality desktop Google Music experience and can't wait for a Clementine implementation, check out [GPMDP](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#readme).

**Comment by rickinbsb** [comment]
@eligrey I`ve been using nuvola player for that, but this seems better, thanks!

**Comment by gmjain** [comment]
+1

**Comment by arvigeus** [comment]
@rickinbsb For some time now GPMDP supports linux as well

**Comment by GregTheHun** [comment]
+1

**Comment by lgaticaq** [comment]
:+1:

**Comment by barakwei** [comment]
+1

**Comment by shoguevara** [comment]
Using https://github.com/radiant-player now, would switch to Clementine if the feature is implemented

**Comment by Yurlungur** [comment]
+1

I would definitely like to see this enhancement.

**Comment by lesthack** [comment]
Or maybe you want to try Google Play Music Desktop Player (Unofficial sure) https://www.googleplaymusicdesktopplayer.com/.

But, I love Clementine player.

**Comment by aldasa** [comment]
+1

**Comment by SemperPeritus** [comment]
+1

**Comment by Dommage** [comment]
+1

**Comment by Fantaser99** [comment]
+1

**Comment by fmanuelsg** [comment]
+1

**Comment by WhiterBlack** [comment]
+1

**Comment by rlkeeney** [comment]
+1

**Comment by z-nut** [comment]
+1

**Comment by ajithrn** [comment]
+1

**Comment by MakarovDi** [comment]
+1

**Comment by Aer0nic** [comment]
+1. using the unofficial desktop app is pretty great but it feels like just a wrapped chromium.

**Comment by RomanPetcube** [comment]
+1

**Comment by nister** [comment]
+1

**Comment by martinblaha** [comment]
+1

**Comment by MirkoLedda** [comment]
+1

**Comment by sonic2kk** [comment]
I did some research and the unofficial API is written in Python. I'm pretty sure there's ways of using the API even if it is in Python with C++, but better yet [there is a C++ port of the library](https://github.com/gwicks/gmusicapi-curl) that uses something called libcurl. I'm sure there's people out there much smarter than me that understand how it works but basically it means that the API works with C++. However this hasn't been updated in about two years, whereas the official version of the API (which is unofficial still) was last updated a month ago at the time of writing.

I did some further research and found a cool project which uses the API to [run Google Play Music in the command line](https://github.com/christopher-dG/gpymusic). But from looking at it I can see that it uses and depends on MPV. To my understanding, Clementine uses GStreamer. How this tool uses MPV is by creating a link and then streaming it through MPV. In my brief tests trying to replicate the behaviour in the program I was unsuccessful. Although as the program clearly works or worked at one point it is definitely possible that by using this API, we can get the behaviour we seek in Clementine. However I can understanding wanting to wait for an official API as this API has updated and broken compatibility with previous versions due to a change with Google Play Music.

tl;dr - API has a C++ port which seems slightly outdated but could be used to give the functionality the people in this issue want, myself included. However as the API is unofficial waiting for an official API (which Google have been reportedly working on for quite some time) seems logical. Having said that, +1. I want this functionality in Clementine and then I can stop using GPMDP.

**Comment by number213** [comment]
+1

**Comment by rokkhan** [comment]
+1

**Comment by felipecock** [comment]
+1

**Comment by clasick** [comment]
+1

**Comment by ardoin** [comment]
+1

**Comment by eric-poitras** [comment]
+2

**Comment by satcom886** [comment]
+1 This would be a killer feature!

**Comment by svyatoslav-kubakh** [comment]
+1

**Comment by Ziiweb** [comment]
+1

**Comment by agent6426** [comment]
+1

**Comment by juanjodarko** [comment]
+1

**Comment by jonny190** [comment]
+1

**Comment by Rasie1** [comment]
+1. Google Music webservice and plugin and gpmdp are all garbage, we need something usable for accessing Google Music storage

**Comment by EugeneGrim** [comment]
+1

**Comment by sschet90** [comment]
R & Linux Mint are my current limits. If Professor Louis Aslett can create a RStudio Amazon Web Services Instance (http://www.louisaslett.com/RStudio_AMI/) that works (?) on iPhones, then would it be possible to create a Clementine AWS Instance that could be used via the AWS iPhone app?

**Comment by Minothor** [comment]
> +1. using the unofficial desktop app is pretty great but it feels like just a wrapped chromium.

Just tried it but it was so laggy and resource intensive that I had to force kill the process to get it to stop playing, uninstalled it in under 3 hours.

I'd love to see a support plugin for Clementine, alas, I fear that it's beyond my knowledge and available time to implement.

**Comment by jonaski** [comment]
There is not an official API for google music so this won't be done. See  #4860

**Comment by skorpions2000** [comment]
+1

**Comment by WebDragon** [comment]
having uploaded my entire CD library to Google Play Music so I could stream it to myself on my android devices when out and about, one would greatly appreciate the additional support for this added to Clementine, either via an option on the Google Drive plugin (since the login would be the same) or via an additional clementine plugin

**Comment by peddy22** [comment]
this would be awesome +1

**Comment by zip9** [comment]
please keep this active +1

**Comment by jason-mehmel** [comment]
+1

**Comment by felipecock** [comment]
+1
